nbs/directives.ipynb

```diff
@@ -0,0 +1,828 @@
+{
+ "cells": [
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "hide_input": false
+   },
+   "outputs": [],
+   "source": [
+    "#|default_exp directives"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "#|hide\n",
+    "from fastcore.test import *\n",
+    "from nbdev.showdoc import show_doc"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Directives\n",
+    "\n",
+    "> The directives available in nbdev. "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "\n",
+    "Directives are special comments that have are preceeded by `#|` that do some kind of pre or post processing of a notebook data before docs are rendered. Some of these directives are part of Quarto, but others are unique to nbdev.\n",
+    "\n",
+    "See the [Quarto docs](https://quarto.org/docs/computations/execution-options.html) for even more execution options.\n",
+    "\n",
+    "Directives can affect the following places:\n",
+    "* the Python module\n",
+    "* the `__all__` export\n",
+    "* the tests\n",
+    "* the documentation"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## `#|default_exp <name>`"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Required for every notebook. This directive names the module where things will be exported. For example:\n",
+    "\n",
+    "```python\n",
+    "#| default_exp baz\n",
+    "\n",
+    "# In a new notebook cell:\n",
+    "\n",
+    "#| export\n",
+    "def my_function(): pass\n",
+    "```\n",
+    "\n",
+    "If our package is named: `bitsnbytes` then we can do:\n",
+    "\n",
+    "```python\n",
+    "from bitsnbytes.baz import my_function\n",
+    "```"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## `#|export`"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Exports the items in the cell into the generated module and documentation. For example:\n",
+    "\n",
+    "```python\n",
+    "#|export\n",
+    "def say_hello(to):\n",
+    "    \"Say hello to somebody\"\n",
+    "    return f'Hello {to}!'\n",
+    "```\n",
+    "\n",
+    "The documentation will look like this:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "#|export\n",
```

Comment:
Remove all `export` directives - we're not exporting a submodule.

Reply:
If you want to "mock" the effect of `#| export` for the docs you can manually call show_doc on it and hide the input.  like this:

```python
#|echo: false
show_doc(say_hello)
```